statistics/estimate.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statistics
+
+import "math"
+
+// calculateEstimateNDV calculates the estimate ndv of a sampled data from a multisize with size total.
+// count[i] stores the count of the i-th element.
+// onlyOnceItems is the number of elements that occurred only once.
+func calculateEstimateNDV(count []uint64, total uint64) (ndv uint64, ratio uint64, onlyOnceItems uint64) {
+	sampleSize := uint64(0)
+	sampleNDV := uint64(len(count))
+	for _, v := range count {
+		if v == 1 {
+			onlyOnceItems++
+		}
+		sampleSize += v
```

Comment:
sampleSize can be obtained in the function caller when build CM-Sketch from samples.